adal/authentication_context.py

```diff
@@ -110,9 +110,9 @@ def options(self):
     def options(self, val):
         self._call_context['options'] = val
 
-    def _acquire_token(self, token_func):
+    def _acquire_token(self, token_func, correlation_id=None):
         self._call_context['log_context'] = log.create_log_context(
-            self.correlation_id, self._call_context.get('enable_pii', False))
+             self.correlation_id or correlation_id, self._call_context.get('enable_pii', False))

```

Comment:
Did you notice the extra leading space here? While it is technically still correct here, but in general, you should make sure your code has exact indention, no more, no less. The current CI/CD check failure is caused by this space too (not by line at 250).

More importantly, should this `or` logic be written as `correlation_id or self.correlation_id`?

Reply:
Fixed the extra space.

It seems to me that it should be self.correlation_id or correlation_id. self.correlation_id is only used if its set by the developer making the calls. In that case, it should default to it, even for device code flow (Both api calls will have the same id). In case self.correlation_id is not set, then correlation_id parameter is used.

Reply:
This can be a judgement call but, wouldn't the whole point of this PR is to ensure the 2 device code requests to have same correlation_id? The proposed `correlation_id or self.correlation_id` is helping customer to achieve that. Even when they setup a self.correlation_id beforehand, I doubt their intention were to say "no I don't want those 2 device code calls to use same correlation_id".

I'm open to different opinion though. We can talk in our standup.